BrainPortal/app/models/custom_filter.rb

```diff
@@ -51,26 +51,98 @@ class CustomFilter < ApplicationRecord
 
   Revision_info=CbrainFileRevision[__FILE__] #:nodoc:
 
+  # A Custom filter have a hash containing the filter parameters
+  # The DATA_PARAMS array is used to withelist these params.
+  DATA_PARAMS =
+  [
+    # Data available on all subclasses
+    :date_attribute,
+    :absolute_or_relative_to,
+    :absolute_or_relative_from,
+    :absolute_or_relative_to,
+    :rel_date_from,
+    :rel_date_to,
+    :abs_from,
+    :abs_to,
+    :relative_from,
+    :absolute_from,
+    :relative_to,
+    :absolute_to,
+    :archiving_status,
+    :user_id,
+    :user,
+
+    # all the array params need to be at the end
+    # of the allowed keys
+    {
+      :type        => [],
+      :status      => [],
+      :sync_status => [],
+      :tag_ids     => [],
+    }
+  ]
+
   include DateRangeRestriction
 
   serialize_as_indifferent_hash :data
 
-  belongs_to    :user
+  belongs_to :user
 
   validates_presence_of   :name
   validates_uniqueness_of :name, :scope  => [:user_id, :type]
   validates_format_of     :name, :with => /\A[\w\-\=\.\+\?\!\s]*\z/,
                                  :message  => 'only the following characters are valid: alphanumeric characters, spaces, _, -, =, +, ., ?, !'
 
-  validate :validate_date
+  validate :valid_self_user
+
+  validate :valid_data_type
+  validate :valid_data_user_id
+  validate :valid_data_archiving_status
+  validate :valid_data_date
+
+  ###############################
+  # Validation of Custom Filter #
+  ###############################
+
+  def valid_self_user #:nodocs:
+    return true if self.user
+    errors.add(:base, 'a custom filter should have a user')
+    return false
+  end
+
+  def valid_data_type #:nodocs:
+    self.data_type = Array(self.data_type).reject { |item| item.blank? }
+    return true if self.data_type.empty?
+    valid_type = self.is_a?(TaskCustomFilter) ? CbrainTask.sti_descendant_names : Userfile.sti_descendant_names
+    return true if ( self.data_type - valid_type ).empty?
+    errors.add(:data_data_type, 'some file type are invalid')
+    return false
+  end
+
+  def valid_data_user_id #:nodocs:
+    return true if self.data_user_id.blank?
+    return true if self.user.available_users.pluck(:id).include? self.data_user_id.to_i
+    errors.add(:data_user_id, 'is not an accessible user')
+    false
+  end
+
+  def valid_data_archiving_status #:nodocs:
+    return true if self.data_archiving_status.blank?
+    valid_status = self.is_a?(TaskCustomFilter) ? ["none", "cluster", "file"] : ["archived", "none"]
+    return true if valid_status.include? self.data_archiving_status
+    errors.add(:data_archiving_status, 'is not a valid archiving status')
+    return false
+  end
+
 
   # Do some validation on the date range filtering
-  def validate_date
+  def valid_data_date #:nodocs:
```

Comment:
this method is still using the old way of accessing data: `self.data["date_attribute"]` instead of `self.data_date_attribute`